adaptdl/adaptdl/torch/data.py

```diff
@@ -195,6 +196,58 @@ def local_bsz_bounds(self):
         """
         return self._local_bsz_bounds
 
+    @property
+    def current_local_bsz(self):
+        """
+        The current logical local batch size used by the dataloader.
+        The batch size returned by the dataloader may be smaller if
+        gradient accumulation is used
+        """
+        return self._current_local_bsz
+
+    @property
+    def accumulation_steps(self):
+        """
+        The number of batches returned by the dataloader before a
+        step is taken.
+        """
+        return self._accumulation_steps
+
+    @property
+    def is_accumulation_step(self):
+        """
+        True iff the current step is a gradient accumulation step:
+        I.e. if the model parameters aren't updated
+        """
+        return self._is_accumulation_step
+
+    @is_accumulation_step.setter
+    def is_accumulation_step(self, value: bool):
```

Comment:
Don't think you need a setter for this since it's only being modified from the same class?

Reply:
Its mostly modified in `parallel.py`. The time its modified in `data.py` is so that the value is correct in the very first step after a `_sync_local_bsz` before it can be updated by `forward` in `parallel.py`.

Reply:
`adascale.py` is treated as the truth for whether the program is in an accumulation step or not, and given that the dataloader needs that in a context manager, its difficult to pass it through without setting a variable.